[BUG]: SIGSEGV when disconnected

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

The CLI seg faults when attempting to upload a contract when disconnected from an endpoint. (This may happen on other commands, not sure)

![cli segv](https://user-images.githubusercontent.com/15834128/158885250-8700d3b7-d153-45e2-baa2-470eaa62e8ab.png)


### Expected behavior

The CLI should not die, but instead return a human readable error message explaining how to fix the problem (e.g. connect to an endpoint).

### Steps to reproduce

1. Open a wallet
2. Attempt to upload a contract

### Environment

```markdown
- OS: Mac OS
```


### Anything else?

_No response_

Copying this from a user on Discord:

I'm trying to register as a block producer, and on the first step, register pob ADDRESS I'm getting a runtime error:

```console
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x81d222]

goroutine 1 [running]:
github.com/koinos/koinos-util-golang/rpc.(*KoinosRPCClient).Call(0x0, {0x94ec13, 0x25}, {0xb017a0, 0xc0000300c0}, {0xb017c0, 0xc0002644b0})
        /home/nathan/go/pkg/mod/github.com/koinos/koinos-util-golang@v0.0.0-20220708183907-67c029f58812/rpc/koinos_rpc_client.go:66 +0xe2
github.com/koinos/koinos-util-golang/rpc.(*KoinosRPCClient).GetContractMeta(0xc0000dc810, {0xc00002c1c0, 0x19, 0x19})
        /home/nathan/go/pkg/mod/github.com/koinos/koinos-util-golang@v0.0.0-20220708183907-67c029f58812/rpc/koinos_rpc_client.go:190 +0xe5
github.com/koinos/koinos-cli/internal/cli.(*RegisterCommand).Execute(0xc000264480, {0x0, 0x300000000}, 0xc0000bd700)
        /home/nathan/src/koinos-cli/internal/cli/contract_commands.go:67 +0x497
github.com/koinos/koinos-cli/internal/cli.(*ParseResults).Interpret(0xc00000e1c8, 0xc00002a240)
        /home/nathan/src/koinos-cli/internal/cli/interpreter.go:311 +0xf3
github.com/koinos/koinos-cli/internal/cli.ParseAndInterpret(0xbf00000005, 0x7f1c030000008a3b, {0xc00002a240, 0x16170f1200})
        /home/nathan/src/koinos-cli/internal/cli/interpreter.go:373 +0x3d6
github.com/koinos/koinos-cli/cmd/cli/interactive.(*KoinosPrompt).executor(0x50fbca, {0xc00002a240, 0xc000130200})
        /home/nathan/src/koinos-cli/cmd/cli/interactive/interactive.go:117 +0x31
github.com/koinos/go-prompt.(*Prompt).Run(0xc0001302d0)
        /home/nathan/go/pkg/mod/github.com/koinos/go-prompt@v0.0.0-20220818181004-5b1028a45a2f/prompt.go:84 +0x765
github.com/koinos/koinos-cli/cmd/cli/interactive.(*KoinosPrompt).Run(0xc00026bc20)
        /home/nathan/src/koinos-cli/cmd/cli/interactive/interactive.go:125 +0xb4
main.main()
        /home/nathan/src/koinos-cli/cmd/cli/main.go:92 +0x42a
```

--- a/internal/cli/commands.go
+++ b/internal/cli/commands.go
@@ -362,10 +362,14 @@ func NewUploadContractCommand(inv *CommandParseResult) Command {
 	}
 }
 
-// Execute calls a contract
+// Execute uploads a contract
 func (c *UploadContractCommand) Execute(ctx context.Context, ee *ExecutionEnvironment) (*ExecutionResult, error) {
 	if !ee.IsWalletOpen() {
 		return nil, fmt.Errorf("%w: cannot upload contract", cliutil.ErrWalletClosed)
+	}
+
+	if !ee.IsOnline() {
+		return nil, fmt.Errorf("%w: cannot upload contract", cliutil.ErrOffline)
 	}
 
 	// Check if the wallet already exists
@@ -712,6 +716,10 @@ func (c *CallCommand) Execute(ctx context.Context, ee *ExecutionEnvironment) (*E
 		return nil, fmt.Errorf("%w: cannot call contract", cliutil.ErrWalletClosed)
 	}
 
+	if !ee.IsOnline() {
+		return nil, fmt.Errorf("%w: cannot call", cliutil.ErrOffline)
+	}
+
 	entryPoint, err := strconv.ParseUint(c.EntryPoint[2:], 16, 32)
 	if err != nil {
 		return nil, err
@@ -900,6 +908,10 @@ func NewReadCommand(inv *CommandParseResult) Command {
 
 // Execute reads from a contract
 func (c *ReadCommand) Execute(ctx context.Context, ee *ExecutionEnvironment) (*ExecutionResult, error) {
+	if !ee.IsOnline() {
+		return nil, fmt.Errorf("%w: cannot read contract", cliutil.ErrOffline)
+	}
+
 	cid := base58.Decode(c.ContractID)
 	if len(cid) == 0 {
 		return nil, errors.New("could not parse contract id")

